src/Logger.h

```diff
@@ -0,0 +1,77 @@
+/* This file is part of the CARTA Image Viewer: https://github.com/CARTAvis/carta-backend
+   Copyright 2018, 2019, 2020, 2021 Academia Sinica Institute of Astronomy and Astrophysics (ASIAA),
+   Associated Universities, Inc. (AUI) and the Inter-University Institute for Data Intensive Astronomy (IDIA)
+   SPDX-License-Identifier: GPL-3.0-or-later
+*/
+
+#ifndef CARTA_BACKEND__LOGGER_H_
+#define CARTA_BACKEND__LOGGER_H_
+
+#include <fmt/format.h>
+
+#define SPDLOG_FMT_EXTERNAL
+#define FMT_HEADER_ONLY
+
+#include <spdlog/sinks/rotating_file_sink.h>
+#include <spdlog/sinks/stdout_color_sinks.h>
+#include <spdlog/spdlog.h>
+
+#include "Constants.h"
+
+enum class LogType { DEBUG, INFO, WARN, ERROR };
+
+void InitLoggers(bool no_log_file, bool debug_log, bool perf_log);
+
+template <typename S, typename... Args>
+void SpdLog(const std::string& log_tag, const LogType& log_type, bool flush_now, const S& format, Args&&... args) {
+    std::shared_ptr<spdlog::logger> logger = spdlog::get(log_tag);
+    if (logger) {
+        switch (log_type) {
+            case LogType::DEBUG:
+                logger->debug(fmt::format(format, args...));
```

Comment:
can you not just pass `format` and `args...` directly into these functions? That way we don't end up actually doing the format call unless that level of debugging is active